src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -53,33 +53,22 @@ public abstract partial class Workspace
 
         private protected void ClearOpenDocuments()
         {
-            List<DocumentId> docIds;
             using (_stateLock.DisposableWait())
             {
-                docIds = _projectToOpenDocumentsMap.Values.SelectMany(x => x).ToList();
-            }
-
-            foreach (var docId in docIds)
-            {
-                this.ClearOpenDocument(docId);
+                foreach (var docId in _projectToOpenDocumentsMap.Values.SelectMany(x => x).ToArray())
```

Comment:
Why ToArray versus ToList? I was always under the impression ToList() was better in this case because you can avoid one final resize of the array which may have to happen in the other case. (No reason to change this PR, but just looked odd.)

Reply:
that's an interesting point.  i generally felt "we don't need resizability, so just go for a simpler type". but it's a good point you make that it might need it to do more work.

Reply:
restored this.